ddtrace/tracer/option.go

```diff
@@ -22,6 +22,11 @@ import (
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/version"
 )
 
+// DefaultServiceName is the service name that will be given to the tracer
+// if one is not configured with WithServiceName or the DD_SERVICE
+// environment variable.
+var DefaultServiceName string = filepath.Base(os.Args[0])
```

Comment:
This is not ok. Global level exported variables can be dangerous, are prone to race conditions and users or foreign (even potentially malicious packages) can alter them. Read more about this theory [here](https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables).

Why do you want to add this?

Reply:
This *was* the way I was exposing the default, so integrations could determine whether the serviceName was the default service name or not. I could have made it a function instead to avoid the global exported var. It is no longer needed since we've slightly changed the design.